internal/app/machined/main.go

```diff
@@ -68,6 +72,30 @@ func sync() {
 	log.Printf("sync hasn't completed in time, aborting...")
 }
 
+func init() {
+	// Explicitly set the default http client transport
+	// to work around our fun proxy.Do once bug.
+	// This is the http.DefaultTransport with the Proxy
+	// func overridden so that the environment variables
+	// with be reread/initialized each time the http call
+	// is made.
+	http.DefaultClient.Transport = &http.Transport{
+		Proxy: func(req *http.Request) (*url.URL, error) {
+			return httpproxy.FromEnvironment().ProxyFunc()(req.URL)
+		},
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
```

Comment:
one more option is to copy from `http.DefaultTransport`, but I guess it's  same in the end (as it's unlikely they're going to change defaults ever)

https://github.com/aptly-dev/aptly/blob/master/http/download.go#L43

Reply:
Probably true, but then again if defaults ever do change we won't have to worry if we copy. Not a blocker for me.